src/Test.UnitTests.Sarif/ExtensionsTests.cs

```diff
@@ -419,9 +419,28 @@ public void Extensions_GetFileName()
             var tuples = new List<Tuple<Uri, string>>
             {
                 new Tuple<Uri, string>(new Uri("file.a", UriKind.Relative), "file.a"),
-                new Tuple<Uri, string>(new Uri("\\file.a", UriKind.Relative), "file.a"),
                 new Tuple<Uri, string>(new Uri("/file.a", UriKind.Relative), "file.a"),
+                new Tuple<Uri, string>(new Uri("\\file.a", UriKind.Relative), "file.a"),
+
+                new Tuple<Uri, string>(new Uri("path/file.a", UriKind.Relative), "file.a"),
+                new Tuple<Uri, string>(new Uri("/path/file.a", UriKind.Relative), "file.a"),
+                new Tuple<Uri, string>(new Uri("path\\file.a", UriKind.Relative), "file.a"),
+                new Tuple<Uri, string>(new Uri("\\path\\file.a", UriKind.Relative), "file.a"),
+
                 new Tuple<Uri, string>(new Uri("file.a?some-query-string", UriKind.Relative), "file.a"),
+                new Tuple<Uri, string>(new Uri("/file.a?some-query-string", UriKind.Relative), "file.a"),
+                new Tuple<Uri, string>(new Uri("\\file.a?some-query-string", UriKind.Relative), "file.a"),
+
+                new Tuple<Uri, string>(new Uri("path/file.a?some-query-string", UriKind.Relative), "file.a"),
+                new Tuple<Uri, string>(new Uri("/path/file.a?some-query-string", UriKind.Relative), "file.a"),
```

Comment:
>file [](http://example.com/codeflow?start=54&length=4)

also consider a mixture of forward and backslashes?

Reply:
added test for mixture of forward and backslashes.